pandas/_libs/join.pyx

```diff
@@ -138,7 +143,13 @@ def left_outer_join(const int64_t[:] left, const int64_t[:] right,
             # this is a short-cut to avoid groupsort_indexer
             # otherwise, the `else` path also works in this case
             rev = np.empty(len(left), dtype=np.intp)
-            rev.put(ensure_platform_int(left_sorter), np.arange(len(left)))
+            rev.put(
+                ensure_platform_int(left_sorter),
+                # NOTE:
+                # this is the C-optimized equivalent of
+                # `np.arange(len(left))`
+                cnp.PyArray_Arange(0, len(left), 1, NPY_INTP),
```

Comment:
What is the reasoning for using a platform int here? We typically use 64 bit

Reply:
```rev``` above is using a dtype of ```np.intp``` and because I used the C-API I needed to supply all arguments, usually the dtype is being infered, but I needed to supply the dtype.

---

Also, if we support 32-bits platform why are we using ```np.int64```, and not ```np.intp```?

(Might be a super dumb question, but I really can't figure this out)

Reply:
You can still get a 64 bit integer on a 32 system; NPY_INT64 is a macro that resolves to the proper type (the C standard also has these in inttypes.h)

I can't speak to entire history but working with fixed width types is generally easier for us